docs/examples/files.md

```diff
@@ -0,0 +1,231 @@
+`pydantic` is a great tool for validating data coming from various sources.
+In this section, we will look at how to validate data from different types of files.
+
+!!! Note:
+    If you're using any of the below file formats to parse configuration / settings, you might want to
+    consider using the [`pydantic-settings`][pydantic_settings] library, which offers builtin
+    support for parsing this type of data.
+
+## JSON data
+
+`.json` files are a common way to store key / value data in a human-readable format.
+Here is an example of a `.json` file:
+
+```json
+{
+    "name": "John Doe",
+    "age": 30,
+    "email": "john@example.com"
+}
+```
+
+To validate this data, we can use a `pydantic` model:
+
+```python test="skip"
+from pydantic import BaseModel, EmailStr, PositiveInt
+
+
+class Person(BaseModel):
+    name: str
+    age: PositiveInt
+    email: EmailStr
+
+
+with open('person.json') as f:
+    json_string = f.read()
+    person = Person.model_validate_json(json_string)
+    print(repr(person))
+    #> Person(name='John Doe', age=30, email='john@example.com')
```

Comment:
I strongly advise using `json_string = pathlib.Path('person.json').read_text()` over `open` `f.read` etc.

The snippet above leaves the file open while validating the json etc. In such a self-contained context and in docs it's not a big deal but it's just not a pattern I would want to encourage users to follow, considering I think using pathlib for reading files is better in 99% of cases. (Well, that's my impression/opinion anyway. Could be convinced otherwise if there are some python authorities claiming you _should_ use open/etc. instead of pathlib for this...)

Reply:
Awesome, thanks for the tip! Good to know 👍